src/setupWorker/setupWorker.ts

```diff
@@ -19,14 +19,15 @@ import { createFallbackStop } from './stop/createFallbackStop'
 import { devUtils } from '../utils/internal/devUtils'
 import { SetupApi } from '../SetupApi'
 import { mergeRight } from '../utils/internal/mergeRight'
+import { SetupWorkerApi as ISetupWorkerApi } from './glossary'
 
 interface Listener {
   target: EventTarget
   eventType: string
   callback: EventListener
 }
 
-export class SetupWorkerApi extends SetupApi<WorkerLifecycleEventsMap> {
+class SetupWorkerApi extends SetupApi<WorkerLifecycleEventsMap> {
```

Comment:
Let's also force this class to implement the `SetupWorkerApi` interface. Without this enforcement, the only area to catch the incongruence is in the `setupWorker` return type, which is rather late. 

```ts
import { SetupWorker } from './glossary'

export class SetupWorkerApi extends SetupApi implements SetupWorker {
  ...
}

export function setupWorker(): SetupWorker { ... }
```

- Rename `SetupWorkerApi` in `glossary` to `SetupWorker`
  - Do the same for `SetupServerApi` 
- Export both `SetupWorker` interface and `SetupWorkerApi` class
- Set `SetupWorker` as the return type for `setupWorker()` to fix the type issue.

Would this approach work? 🤔

Reply:
I agree with the implements part, I hesitated to do that.

But as I said above the `SetupWorkerApi` interface is already exposed, so we should not change its name to avoid a breaking change.

The class was only exposed through `setupWorker` return type. And it cannot be used due to those TS declaration file limitations.

Tell me what you think and I'll update the PR

Reply:
Makes sense. Let's split this change in two then:

1. Import `SetupWorkerApi` type, alias it to `SetupWorker`, adjust the return type of `setupWorker` function. Release as a fix.
2. Open a follow-up that exports the `SetupWorkerApi` (class) and renames the `SetupWorkerApi` (interface) in `glossary.ts` to `SetupWorker`. Do the same for `SetupServer*`. Release a breaking change. 

What do you think?

Reply:
Seems good. I had to check, but in fact exporting the class instead of the interface removes the TS error (I was expecting to still have it)

Reply:
I updated this one, I try to open the second one on top of this one